### PR TITLE
test(mysql): initiate mysql integration tests

### PR DESCRIPTION
**Pull request by MrSwitch**
Initial MySQL Integration tests

### CODE REVIEW

File: test/integration/mocha-hooks/mysql.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+module.exports.mochaHooks = {
+	async beforeAll() {
+
+		// BeforeAll happens per-process/thread, so each subsequent test can reset the db without it interfering with other tests in that thread
+		this.timeout(5000);
+		// To support parallel tests, we create a separate db per-process/thread and just reset the state (truncate tables) per-test.
+		const testDb = await createNewDb(defaultDbSuffix);
+
+
+		const db = require('../helpers/db');
```
Comment (MrSwitch) [inline test/integration/mocha-hooks/mysql.js:104]: `../helpers/db` is reloaded everytime, so i have to re-require it when in --watch mode. 😭

File: test/integration/run.sh [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	echo "count $i";
+    sleep 2
+    if [[ "$i" -gt '20' ]]; then
+      echo "${dep} failed to start. Final status: $(docker inspect --format='{{.State.Health.Status}}' "dare_${dep}")"
+      docker-compose down -v -t 6
+      exit 1
+    fi
+  done
+  echo "${dep} up"
+  sleep 5
```
Comment (MrSwitch) [inline test/integration/run.sh:67]: MySQL isn't always available straight away so wait another 5 seconds 😭

File: test/integration/mocha-hooks/mysql.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
+/* eslint-disable no-console */
+const {execSync} = require('child_process');
+const mysql = require('mysql2/promise');
+const fs = require('fs');
+
+/*
+ * Mocha deletes the require cache, it will create a new instance of helpers/db on each run of the tests
```
Comment (sadams) [inline test/integration/mocha-hooks/mysql.js:7]: Nit: Might help anyone who tries to fix this in the future to specify this happens in  mocha watch mode only

**Review by sadams — APPROVED** [review]
Nice work! 👍

**Pull request merged**